### ISSUE TITLE
Create aliases for colors

### ISSUE THREAD

**Issue by Noodlez1232** [danth/stylix#207]
In the [style guide](https://danth.github.io/stylix/styling.html) there's some guidelines for what colors to use for what. I think it'd be pretty useful to add aliases for these uses under `stylix.colors.general.{colorAlias}` `stylix.colors.notifications.{colorAlias}` and so on.

With this, we might be able to have modules that look like this (using the bemenu module because it's nice and simple)

```nix
# KEY
with config.lib.stylix.colors.withHashtag;
with config.lib.stylix.colors.withHashtag.general;
  config = lib.mkMerge [
    (lib.mkIf config.stylix.targets.bemenu.enable {
      programs.bemenu.settings = with config.stylix.targets.bemenu; {
        tb = "${defaultBackground}${bemenuOpacity}"; # Title bg
        nb = "${defaultBackground}${bemenuOpacity}"; # Normal bg
        fb = "${defaultBackground}${bemenuOpacity}"; # Filter bg
        hb = "${selectionBackground}${bemenuOpacity}"; # Highlighted bg
        sb = "${selectionBackground}${bemenuOpacity}"; # Selected bg
        scb = "${defaultBackground}"; # Scrollbar bg

        hf = "${warning}"; # Highlighted fg
        sf = "${defaultText}"; # Selected fg
        tf = "${defaultText}"; # Title fg
        ff = "${defaultText}"; # Filter fg
        nf = "${defaultText}"; # Normal fg
        scf = "${selectionBackground}"; # Scrollbar fg

        ab = "${if alternate then alternateBackground else defaultBackground}"; # Alternate bg
        af = "${if alternate then alternateText else defaultText}"; # Alternate fg

        # Font name
        fn = "${sansSerif.name} ${lib.optionalString (fontSize != null) (builtins.toString fontSize)}";
      };
    })
  ];
```

It's much easier to read what's going on, and could make styling much easier to unify, since you could just use the color names instead of having to go through the documentation.

**Comment by danth** [comment]
This looks very nice, and it's something I've been considering for a while.

It might be useful for implementing #205 since that will introduce new named colors which lie outside of the main palette.

It also opens up the possibility of supporting non-base16 color schemes in the future.

Internally, palettes are handled by [base16.nix](https://github.com/SenchoPens/base16.nix) so any modifications would need to go through there, or more likely we would replace it with our own library since this is such a large change.

**Comment by Noodlez1232** [comment]
That's true, which might mean a break from any other base16.nix compatible extensions that stylix uses.
I kinda wonder if we could store these somewhere else, then.
For example store it in `lib.stylix.colorAliases` *for now* until we can get some sort of other solution worked out

**Comment by danth** [comment]
Note that that would take away the ability to use the colors in mustache templates, which may or may not be something we want to support. Theoretically the template files could just be converted to multiline strings within Nix, using `${...}` in place of `{{...}}`.
